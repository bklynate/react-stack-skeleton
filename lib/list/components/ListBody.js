"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = ListBody;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * The wrapper of the todo list. The children must be <li />s.
 */
function ListBody(_ref) {
  var onClickCompleteAllItems = _ref.onClickCompleteAllItems,
      children = _ref.children,
      otherProps = _objectWithoutProperties(_ref, ["onClickCompleteAllItems", "children"]);

  return _react2.default.createElement(
    "section",
    { className: "main" },
    _react2.default.createElement("input", { className: "toggle-all", type: "checkbox", onChange: onClickCompleteAllItems }),
    _react2.default.createElement(
      "label",
      { htmlFor: "toggle-all" },
      "Mark all as complete"
    ),
    _react2.default.createElement(
      "ul",
      { className: "todo-list" },
      children
    )
  );
}