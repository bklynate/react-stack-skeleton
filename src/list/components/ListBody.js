/* @flow */
import React from 'react';

type Props = {
  onClickCompleteAllItems: (e: SyntheticInputEvent) => void,
  children?: any,
};

/**
 * The wrapper of the todo list. The children must be <li />s.
 */
export default function ListBody({ onClickCompleteAllItems, children, ...otherProps}: *): React.Element<*> {
  return (
    <section className="main">
      <input className="toggle-all" type="checkbox" onChange={onClickCompleteAllItems} />
      <label htmlFor="toggle-all">Mark all as complete</label>
      <ul className="todo-list">
        {children}
      </ul>
    </section>
  );
}
