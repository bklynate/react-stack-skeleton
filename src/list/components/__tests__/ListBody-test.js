import React from 'react';
import renderer from 'react-test-renderer';

import ListBody from '../ListBody';

test('it displays the children in the list.', () => {
  const component = renderer.create(
    <ListBody>
      <li>Test Item 0</li>
      <li>Test Item 1</li>
      <li>Test Item 2</li>
    </ListBody>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls the onChangeNewItem function if the item is changed.', () => {
  const onClickFn = jest.fn();
  const component = renderer.create(
    <ListBody onClickCompleteAllItems={onClickFn} />
  );
  const tree = component.toJSON();
  const selectAllInput = tree.children[0];

  selectAllInput.props.onChange();
  expect(onClickFn).toHaveBeenCalledTimes(1);
});
