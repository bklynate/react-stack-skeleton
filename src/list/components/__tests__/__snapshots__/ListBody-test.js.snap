exports[`test it displays the children in the list. 1`] = `
<section
  className="main">
  <input
    className="toggle-all"
    onChange={undefined}
    type="checkbox" />
  <label
    htmlFor="toggle-all">
    Mark all as complete
  </label>
  <ul
    className="todo-list">
    <li>
      Test Item 0
    </li>
    <li>
      Test Item 1
    </li>
    <li>
      Test Item 2
    </li>
  </ul>
</section>
`;
